src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <CPP/custom_gpu_primitive.hpp>
+#include <CPP/reshape.hpp>
+
+#include "ngraph/runtime/intelgpu/intelgpu_layout.hpp"
+#include "ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.hpp"
+
+#include "ngraph/util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static vector<cldnn_arg> parameters_2inp_1out = {{arg_input, 0}, {arg_input, 1}, {arg_output, 0}};
+
+static string array_dims(const Shape& dimentions)
+{
+    string buffer;
+
+    for (auto const& dim : dimentions)
+    {
+        buffer += "[" + to_string(dim) + "]";
+    }
+
+    return buffer;
+}
+
+static string access_dims(const Shape& dimentions, const AxisSet& axis = {})
+{
+    size_t var_idx = 0;
+    string buffer;
+
+    for (auto i = dimentions.cbegin(); i != dimentions.cend(); ++i, ++var_idx)
+    {
+        if (axis.find(var_idx) == axis.end())
+        {
+            buffer += "[i" + to_string(var_idx) + "]";
+        }
+    }
+
+    return buffer;
+}
+
+static string
+    access_dims_strided(const Shape& dimentions, const Shape& pad_below, const Shape& pad_interior)
+{
+    string buffer;
+    size_t var_idx = 0;
+
+    for (auto i = dimentions.cbegin(); i != dimentions.cend(); ++i, ++var_idx)
+    {
+        buffer += "[i" + to_string(var_idx) + " * (" + to_string(pad_interior.at(var_idx)) +
+                  " + 1) + " + to_string(pad_below.at(var_idx)) + "]";
+    }
+
+    return buffer;
+}
+
+void runtime::intelgpu::do_pad_kernel(cldnn::topology& topology,
+                                      const string& input_name,
+                                      const Shape& input_shape,
+                                      const string& scalar_name,
+                                      const string& output_name,
+                                      const Shape& output_shape,
+                                      const element::Type& output_type,
+                                      const Shape& pad_below,
+                                      const Shape& pad_interior)
+{
+    const size_t input_count = shape_size<Shape>(output_shape);
+    const string entry_point_name = "op_pad_kernel";
+    ostringstream kernel_code;
```

Comment:
In the codegen library there is a CodeWriter class which makes emitting nice, readable code easy. It does indenting and has helpters for code block begin and end.
https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/codegen/code_writer.hpp
It is just a stringstream with some helpers.